### ISSUE TITLE
Syslog notifier produces CEF events with wrong field mapping

### ISSUE THREAD

**Issue by yrro** [stackrox/stackrox#5400]
The fields of a CEF message are:

```
CEF:Version|Device Vendor|Device Product|Device Version|Signature ID|Name|Severity|Extension
```

But the [CEF message formatting code](https://github.com/stackrox/stackrox/blob/81da9e0c5aef4ee9b228c476c850f6aa676e117e/central/notifiers/syslog/syslog.go#L195) generates messages with a different field mapping:

```go
func alertToCEF(alert *storage.Alert, notifier *storage.Notifier) string {
   ...
   return getCEFHeaderWithExtension("Alert", alert.GetPolicy().GetName(), severity, makeExtensionFromPairs(extensionList))
}

func getCEFHeaderWithExtension(deviceEventClassID, name string, severity int, extension string) string {
	return fmt.Sprintf("CEF:0|StackRox|Kubernetes Security Platform|%s|%s|%d|%s|%s", version.GetMainVersion(), deviceEventClassID, severity, name, extension)
}
```

Note that `deviceEventClassID` is hard coded to the string `Alert`; I think the policy ID could be more appropriate: the corresponding CEF field is called `Signature ID`.

CEF field | Stackrox behaviour | Desired behaviour
| - | - | - |
Signature ID | `Alert` | `Policy.{`policy id`}`
Name |  severity | name
Severity | name | severity

As a result the messages stored in SIEMs aren't so useful since the Name and Severity fields are swapped around and we can't query/filter for policy ID.

Here's what we see in Microsoft Sentinel:

![image](https://user-images.githubusercontent.com/309468/228012678-3b4fe7d2-498f-4f87-99f1-a872e1a2b37d.png)

**Comment by theencee** [comment]
Unfortunately we have customers relying on this current order so we'll have to deprecate it and/or come up with a way to choose an alternative format. This is on my backlog and will get back soon.

**Comment by yrro** [comment]
From this user's PoV a menu option to let me pick 'CEF' or 'CEF (legacy field order)' would do the job. Set to the legacy option on upgrades and defaulting to 'CEF' for newly created integrations.

**Comment by theencee** [comment]
Closing as we just released 4.2.0 which has this fix
